backend/src/error.rs

```diff
@@ -0,0 +1,46 @@
+use actix_web::{http::StatusCode, HttpResponse};
+use serde_json::error::Category;
+use subxt::ext::sp_core::crypto::SecretStringError;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum AppError {
+    #[error("Database error: {0}")]
+    Database(#[from] sqlx::Error),
+    #[error("Blockchain error: {0}")]
+    Subxt(#[from] subxt::Error),
+    #[error("Server error: {0}")]
+    ActixWeb(#[from] actix_web::Error),
+    #[error("Signature error: {0}")]
+    Secret(#[from] SecretStringError),
+    #[error("JSON error: {0}")]
+    Json(#[from] serde_json::Error),
+    #[error("Hex error: {0}")]
+    Hex(#[from] hex::FromHexError),
+}
+
+impl actix_web::error::ResponseError for AppError {
+    fn error_response(&self) -> HttpResponse {
+        HttpResponse::build(self.status_code()).body(self.to_string())
+    }
+
+    fn status_code(&self) -> StatusCode {
+        match self {
+            AppError::Database(e) => match e {
+                sqlx::Error::RowNotFound => StatusCode::NOT_FOUND,
+                _ => StatusCode::INTERNAL_SERVER_ERROR,
+            },
+            AppError::Hex(e) => match e {
+                hex::FromHexError::InvalidHexCharacter { .. } => StatusCode::BAD_REQUEST,
+                hex::FromHexError::InvalidStringLength => StatusCode::BAD_REQUEST,
+                _ => StatusCode::INTERNAL_SERVER_ERROR,
+            },
```

Comment:
I think at least for these options I would not use a nested match.
```suggestion
            AppError::Database(sqlx::Error::RowNotFound) => StatusCode::NOT_FOUND,
            AppError::Hex(hex::FromHexError::InvalidHexCharacter { .. }) => StatusCode::BAD_REQUEST,
            AppError::Hex(hex::FromHexError::InvalidStringLength) => StatusCode::BAD_REQUEST,
```

Reply:
made it less nested: [b8ef7d5](https://github.com/KILTprotocol/attester/pull/1/commits/b8ef7d5a2db19346a95d5c7e76ec857e773e95d0)